packages/server/lib/util/require_async.ts

```diff
@@ -0,0 +1,78 @@
+import _ from 'lodash'
+import * as path from 'path'
+import * as cp from 'child_process'
+import * as inspector from 'inspector'
+import * as util from '../plugins/util'
+import * as errors from '../errors'
+import Debug from 'debug'
+
+const debug = Debug('cypress:server:require_async')
+
+let requireProcess: cp.ChildProcess | null
+
+interface RequireAsyncOptions{
+    projectRoot: string
+    loadErrorCode: string
+    /**
+     * members of the object returned that are functions and will need to be wrapped
+     */
+    functionNames: string[]
+}
+
+interface ChildOptions{
+    stdio: 'inherit'
+    execArgv?: string[]
+}
+
+const killChildProcess = () => {
+  requireProcess && requireProcess.kill()
+  requireProcess = null
+}
+
+export default async function requireAsync (filePath: string, options: RequireAsyncOptions): Promise<any> {
```

Comment:
```suggestion
export default async function requireAsync (filePath: string, options: RequireAsyncOptions): Promise<void> {
```

This function will never return a value.

Reply:
I think it will return a value. 
This is the purpose of the `resolve()` function.

It will return the exported object of the file.
Since we do not know the type of that value we say `any`.

I could also go for `T=any`. If we know the theoretical contents of a file this could be useful. Though in this case I would prefer the use of `as`